src/components/StarRating/StarRating.tsx

```diff
@@ -56,7 +56,7 @@ export class StarRating extends React.PureComponent<any, { rating }> {
         }
     }
 
-    UNSAFE_componentWillReceiveProps(nextProps) {
+    componentDidUpdate(nextProps) {
```

Comment:
Lets change the variable name here to be `oldProps` or whatever, for the sake of future editors :)